### DISCUSSION TITLE
Can the Shader use outline in URP Render Mode?

### DISCUSSION THREAD

**Discussion by DavidWeng916** [pixiv/vroid-sdk-developers#109] [Q&A] [answered]
Hi, I have a question, I've been working on how to make VRM models have an Outline effect in UPR rendering mode.

I have reference [106](https://github.com/pixiv/vroid-sdk-developers/discussions/106) which changes ModelLoader.LoadVrm to the following
`ModelLoader.LoadVrm(... , materialGenerator: new UrpVrm10MaterialDescriptorGenerator());`

I'm trying to test the Example.scene in VRoidSDK_v0.3.0. I see that the `VRM10/Universal Render Pipeline/MToon10` Shader seems to have the Outline function, but I don't see any Outline effect.
<img width="1064" alt="截圖 2023-09-09 下午7 48 57" src="https://github.com/pixiv/vroid-sdk-developers/assets/115971193/489d8326-a04c-4475-adb0-f180fd49f7da">
<img width="1064" alt="截圖 2023-09-09 下午7 36 12" src="https://github.com/pixiv/vroid-sdk-developers/assets/115971193/fce546ad-5b07-4666-9d70-75caf6856935">

I would like to ask how to turn on the Outline effect?
Or do you have any suggestions on how to enable the Outline effect for VRM models in URP rendering mode?

My Unity version is 2021.3.20f1

**Comment by nyamadan** [comment]
I'm sorry for the late reply.

Did you install the URP package into your project?

https://docs.unity3d.com/Packages/com.unity.render-pipelines.universal@14.0/manual/InstallURPIntoAProject.html

Simply adding an argument to a method does not enable URP.

**Reply by DavidWeng916** [reply]
Hi, @nyamadan, Yes, I have installed the URP package. But I don't see the effect of the outline.
<img width="1046" alt="截圖 2023-09-13 下午1 36 38" src="https://github.com/pixiv/vroid-sdk-developers/assets/115971193/5ec55404-a9d6-4442-a779-e2568b21b1be">

**Reply by nyamadan** [reply, marked as answer]
We found a way to render outlines.

You can render the outline by adding the `MToon Outline Render Feature` to the `Universal Render Data` asset.

![RenderFeature](https://github.com/pixiv/vroid-sdk-developers/assets/304238/232265cc-8329-4f4c-b08d-48c8f17c1bba)

**Reply by DavidWeng916** [reply]
Hi, @nyamadan I was busy with other work and I apologize for not having the chance to test it until now, I followed your instructions and the outline effect is showing up correctly, thank you very much.
But I found another problem, I found that with VRoid 0.3.0 some models' hair doesn't flow anymore, I have to open another [discussion](https://github.com/pixiv/vroid-sdk-developers/discussions/127).
